service/resource/cloudformation/desired.go

```diff
@@ -0,0 +1,22 @@
+package cloudformation
+
+import (
+	"context"
+
+	"github.com/giantswarm/aws-operator/service/key"
+	"github.com/giantswarm/microerror"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	mainStack, err := newMainStack(customObject)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return mainStack, nil
+}
```

Comment:
Some general questions to sort out where we are going to. In past implementations of operators we struggled with finding out what the actual state of a resource being reconciled should be. I want to prevent a lot of back and forth and get on the same page. One requirement for the framework is that the state being dispatched between methods should always be the same. Here we assume the desired state is `*awsCF.CreateStackInput`. Are we sure `*awsCF.CreateStackInput` is also what is returned by `GetCurrentState` and that `*awsCF.CreateStackInput` is what gets provided to `Apply*Change` methods? Do our dependency clients we rely on allow to fetch something as `*awsCF.CreateStackInput` or can we make up the current state? Do our dependency clients we rely on allow to update or delete state of this type?

Reply:
Yes, good point the state object is very important. For the azure-operator we defined our own type for the Deployment which is pretty much equivalent to the Cloud Formation resource here.

@fgimenez Lets sync on this tomorrow.

Reply:
> Are we sure *awsCF.CreateStackInput is also what is returned by GetCurrentState and that *awsCF.CreateStackInput is what gets provided to Apply*Change methods?

Nott at all! :) Thx a lot makes total sense, I've pushed changes to use a `Stack` type where we can put the required fields, pls take another look.

> @fgimenez Lets sync on this tomorrow.

@rossf7 sure, we need to talk too about how to compare states.